utils.go

```diff
@@ -248,7 +264,6 @@ func (a testAddr) String() string {
 }
 
 type testConn struct {
-	net.Conn
```

Comment:
not 100% sure about this change, but since we're running tests, it shouldn't be a problem

Reply:
okay had a talk with kiyonlin, testConn already implement all the needed stuff, so net.Conn should be not necassary